### PR TITLE
Article style refactoring

### PR DESCRIPTION
**Pull request by olcplif**
- [x] Update style in admins/articles according to styles admins/users.
- [x] Added author email popup on the article index page (for admins).
- [x] Added preview text and image on the articles index page.
- [x] Added 'accordion' style for the small screen (articles index page for admin).

[https://trello.com/c/yotBfiUK/80-apply-admin-styles-to-article-pages](https://trello.com/c/yotBfiUK/80-apply-admin-styles-to-article-pages)

### CODE REVIEW

File: app/javascript/packs/accordions.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
+(function ($) {
+  "use strict";
+  $.fn.responsiveTable = function() { 
+
+    var toggleColumns = function($table) {
```
Comment (viktoriia-horobets) [inline app/javascript/packs/accordions.js:5]: please, use 'let' instead of 'var'

Comment (olcplif) [reply app/javascript/packs/accordions.js]: fixed

**Review by bagiratania98 — COMMENTED** [review]
Lgtm, but we said that we won't use jquery or maybe I'm wrong?

File: app/views/admins/articles/index.html.erb [hunk 2]
```diff
@@ -35,13 +35,22 @@
...
             <% end %>
             <%= link_to admins_article_path(article), method: :delete, data: { confirm: "Are you sure?" }, class: 'button-functions-articles' do %>
-              <span><i class='material-icons'>delete</i><%= t('admins.articles.views.buttons.delete') %></span>
+              <span>
+                <i class='material-icons'>delete</i>
+                <span><%= t('admins.articles.views.buttons.delete') %></span>
+              </span>
             <% end %>
           </div>
         </div>
```
Comment (stasfly) [inline app/views/admins/articles/index.html.erb:56]: Please add an extra empty line in the end of every single file you did.

**Review by stasfly — APPROVED** [review]

**Review by mmrshk — APPROVED** [review]
lgtm

File: app/views/admins/articles/index.html.erb (hunk 2 above)
Comment (olcplif) [reply app/views/admins/articles/index.html.erb:56]: Sorry, but I can't find files without empty lines in the end. Maybe you can point me to one of them?

File: app/assets/stylesheets/admin.scss [hunk 3]
```diff
@@ -324,7 +324,24 @@ select {
...
+    background-color: $admin-background-color-button;
+    font-size: 18px;
+    border-radius: 30px;
+    padding: 10px;
+    text-align: center;
+  }
+}
+
+.add-features {
+  display: flex;
```
Comment (MishaKlymenko) [inline app/assets/stylesheets/admin.scss:336]: add flex-direction or flex-flow

File: app/assets/stylesheets/admins/article/_accordion.scss [hunk 4]
```diff
@@ -0,0 +1,52 @@
+$accordionColour: $admin-background-color-button;
+$iconSize: 32px;
+$breakpoint: 800px;
+
+.accordion {
+  position: relative; top: -3px; left: -3px;
```
Comment (MishaKlymenko) [inline app/assets/stylesheets/admins/article/_accordion.scss:6]: add new line for top and left

File: app/assets/stylesheets/admins/article/_accordion.scss [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+  cursor: pointer;
+
+  @media all and (min-width: $breakpoint) { display: none; }
+  &:hover, &:focus {
+    outline: none;
+    background-color: $accordionColour;
+    color: $white-color;
+  }
+
+  &[aria-selected="true"] {
```
Comment (MishaKlymenko) [inline app/assets/stylesheets/admins/article/_accordion.scss:27]: remove this style

File: app/assets/stylesheets/admins/article/_accordion.scss [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+    top: 50%;
+    left: 0.3em;
+    margin-top: -($iconSize/1.75);
+    font-weight: normal;
+    font-size: $iconSize;
+    line-height: $iconSize;
+    background-size: $iconSize $iconSize;
+  }
+
+  &[aria-selected="true"]:before {
```
Comment (MishaKlymenko) [inline app/assets/stylesheets/admins/article/_accordion.scss:49]: chenge rule

File: app/assets/stylesheets/admins/article/_index_table.scss [hunk 7]
```diff
@@ -0,0 +1,183 @@
...
+  display: table-cell;
+  vertical-align: middle;
+
+  // .button-functions-articles:first-child {
+  //   // padding-left: 6px;
+  // }
+
+  .button-functions-articles {
+    display: flex;
+    // flex-direction: column;
```
Comment (MishaKlymenko) [inline app/assets/stylesheets/admins/article/_index_table.scss:112]: remove this comment

File: app/assets/stylesheets/admins/article/_index_table.scss [hunk 8]
```diff
@@ -0,0 +1,183 @@
...
+  
+    span {
+      display: flex;
+      flex-direction: column;
+      text-align: center;
+      align-items: center;
+    }
+
+    @media screen and (max-width: 1400px) {
+      // flex-direction: column;
```
Comment (MishaKlymenko) [inline app/assets/stylesheets/admins/article/_index_table.scss:130]: remove comments

File: app/javascript/packs/accordions.js [hunk 9]
```diff
@@ -0,0 +1,28 @@
+(function ($) {
```
Comment (MishaKlymenko) [inline app/javascript/packs/accordions.js:1]: remove this file

File: app/javascript/packs/application.js [hunk 10]
```diff
@@ -2,10 +2,13 @@ require("@rails/ujs").start()
 import "@hotwired/turbo-rails";
 require("@rails/activestorage").start()
 require("channels")
-import 'materialize-css/dist/js/materialize'
 
+
+require('jquery')
```
Comment (MishaKlymenko) [inline app/javascript/packs/application.js:7]: remove

**Review by MishaKlymenko — CHANGES_REQUESTED** [review]

File: app/assets/stylesheets/admins/article/_accordion.scss (hunk 4 above)
Comment (olcplif) [reply app/assets/stylesheets/admins/article/_accordion.scss:6]: fixed

File: app/assets/stylesheets/admins/article/_accordion.scss (hunk 5 above)
Comment (olcplif) [reply app/assets/stylesheets/admins/article/_accordion.scss:27]: updated

File: app/assets/stylesheets/admins/article/_accordion.scss (hunk 6 above)
Comment (olcplif) [reply app/assets/stylesheets/admins/article/_accordion.scss:49]: updated

File: app/assets/stylesheets/admins/article/_index_table.scss (hunk 7 above)
Comment (olcplif) [reply app/assets/stylesheets/admins/article/_index_table.scss:112]: fixed

File: app/assets/stylesheets/admins/article/_index_table.scss (hunk 8 above)
Comment (olcplif) [reply app/assets/stylesheets/admins/article/_index_table.scss:130]: fixed

File: app/javascript/packs/accordions.js (hunk 9 above)
Comment (olcplif) [reply app/javascript/packs/accordions.js:1]: removed

File: app/javascript/packs/application.js (hunk 10 above)
Comment (olcplif) [reply app/javascript/packs/application.js:7]: fixed

File: app/assets/stylesheets/admins/article/_accordion.scss [hunk 11]
```diff
@@ -0,0 +1,68 @@
...
+.js-table-accordions {
+  .cell {
+    @media screen and (max-width: $breakpoint) {
+      display: none;
+    }
+  }
+
+  .accordion {
+    position: relative;
+    top: -3px;
```
Comment (MishaKlymenko) [inline app/assets/stylesheets/admins/article/_accordion.scss:14]: why the negative values?

File: app/assets/stylesheets/admins/article/_index_table.scss [hunk 12]
```diff
@@ -0,0 +1,207 @@
...
+    width: 24%;
+    padding: 6px 30px 6px 12px;
+    border-bottom-right-radius: 30px; 
+    border-top-right-radius: 30px; 
+  }
+
+  .header-cell {
+    display: table-cell;
+    vertical-align: middle;
+    padding: 6px 12px 6px 12px;
```
Comment (MishaKlymenko) [inline app/assets/stylesheets/admins/article/_index_table.scss:49]: change to padding: 6px 12px; Reading documentation

File: app/assets/stylesheets/admins/article/_index_table.scss [hunk 13]
```diff
@@ -0,0 +1,207 @@
...
+.row-table {
+  display: table-row;
+  min-height: 60px;
+
+  @media screen and (max-width: $breakpoint) { 
+    & {
+      display: flex;
+      flex-flow: column nowrap;
+      width: 100%;
+      padding: 6px 30px 6px 30px;
```
Comment (MishaKlymenko) [inline app/assets/stylesheets/admins/article/_index_table.scss:79]: change to padding: 6px 30px; Reading documentation

File: app/assets/stylesheets/admins/article/_index_table.scss [hunk 14]
```diff
@@ -0,0 +1,207 @@
...
+    white-space: normal;
+  }
+}
+
+.cell:last-child {
+  padding: 6px 30px 6px 12px;
+}
+
+.cell {
+  padding: 6px 12px 6px 12px;
```
Comment (MishaKlymenko) [inline app/assets/stylesheets/admins/article/_index_table.scss:114]: change to padding: 6px 12px; Reading documentation

File: app/javascript/packs/accordions.js [hunk 15]
```diff
@@ -0,0 +1,10 @@
+let accordions = document.querySelectorAll('.js-table-accordions');
+accordions.forEach(accordion => {
+  accordion.addEventListener('click', (event) => {
+    table = accordion.parentElement.querySelectorAll('.js-table-accordions.active');
```
Comment (MishaKlymenko) [inline app/javascript/packs/accordions.js:4]: tables

**Review by MishaKlymenko — CHANGES_REQUESTED** [review]

File: app/assets/stylesheets/admins/article/_accordion.scss (hunk 11 above)
Comment (olcplif) [reply app/assets/stylesheets/admins/article/_accordion.scss:14]: this is a mistake. corrected

File: app/javascript/packs/accordions.js (hunk 15 above)
Comment (olcplif) [reply app/javascript/packs/accordions.js]: corrected

File: app/assets/stylesheets/admins/article/_index_table.scss (hunk 13 above)
Comment (olcplif) [reply app/assets/stylesheets/admins/article/_index_table.scss]: corrected

File: app/assets/stylesheets/admins/article/_index_table.scss (hunk 14 above)
Comment (olcplif) [reply app/assets/stylesheets/admins/article/_index_table.scss]: corrected

File: app/assets/stylesheets/admins/article/_index_table.scss (hunk 12 above)
Comment (olcplif) [reply app/assets/stylesheets/admins/article/_index_table.scss]: corrected

**Review by MishaKlymenko — APPROVED** [review]

**Pull request merged**